graviti/dataframe/frame.py

```diff
@@ -113,11 +122,80 @@ def __setitem__(self, key: str, value: Iterable[Any]) -> None:
         pass
 
     def __repr__(self) -> str:
-        return f"\n DataFrame\n columns: {self._columns} \n index: {self._index}"
+        header_info, data, _ = self._get_repr_header(self._columns)
+        repr_length = [_DEFAULT_REPR_NUMBER] * (len(data) + 1)
+
+        repr_table: List[Union[List[int], List[str]]] = [
+            [_PLACEHOLDER] * (len(data) + 1) for _ in range(len(header_info))
+        ]
+        for index, value in header_info.items():
+            for name, info in value.items():
+                repr_table[index][info["start"] + 1] = name
+                repr_length[info["start"] + 1] = max(len(name), repr_length[info["start"] + 1])
+
+        for index in self._get_repr_indices():
+            cur_line = [index]
+            for idx, column in enumerate(data):
+                repr_length[idx + 1] = max(len(column[index].__format__("")), repr_length[idx + 1])
+                cur_line.append(column[index])
+            repr_table.append(cur_line)
+
+        repr_lines = [
+            "".join(f"{item:<{repr_length[index]+1}}" for index, item in enumerate(line))
+            for line in repr_table
+        ]
+        if self.__len__() > _DEFAULT_REPR_NUMBER:
+            repr_lines.append(f"...({self.__len__()})")
+        return "\n".join(repr_lines)
+
+    def __format__(self, format_spec: str) -> str:
```

Comment:
why using `__format__`???

Reply:
repr use `__format__` to display nested Dataframe.

Reply:
will it breaks the normal `format` method?